services/vision/mlvision/detector.go

```diff
@@ -47,42 +49,35 @@ func attemptToBuildDetector(mlm mlmodel.Service) (objectdetection.Detector, erro
 	return func(ctx context.Context, img image.Image) ([]objectdetection.Detection, error) {
 		origW, origH := img.Bounds().Dx(), img.Bounds().Dy()
 		resized := resize.Resize(inWidth, inHeight, img, resize.Bilinear)
-		inMap := make(map[string]interface{})
+		inMap := ml.Tensors{}
 		switch inType {
 		case UInt8:
-			inMap["image"] = rimage.ImageToUInt8Buffer(resized)
+			inMap["image"] = tensor.New(tensor.WithBacking(rimage.ImageToUInt8Buffer(resized)))
```

Comment:
Should this be setting some shape information for the tensor?

Reply:
So, there is where order matters! In tensor land, the order is always (HEIGHT,WIDTH,3), in image land it's always (WIDTH,HEIGHT,3). I will assume that the shape will be (HEIGHT,WIDTH,3), but it's always possible for the server to reshape to the input how it sees fit.